src/vm/methodtablebuilder.cpp

```diff
@@ -1526,6 +1526,7 @@ MethodTableBuilder::BuildMethodTableThrowing(
             {
                 // Disable AOT compiling for managed implementation of hardware intrinsics.
                 // We specially treat them here to ensure correct ISA features are set during compilation
+                // We can allow these to AOT compile in CoreLib since CoreLib versions with the runtime.
```

Comment:
CoreLib versioning with the runtime is the reason why have don't throw for Vector64/128/256 use in CoreLib here:

https://github.com/dotnet/coreclr/blob/b18bb0b432da925f71c266ed5d17f54c0d202f27/src/vm/methodtablebuilder.cpp#L9582-L9587

It's not the reason in this spot.

I don't recall the reason for this one but you can probably find it in the 100+ comments on #24689 and #24917 if you want. I think it was just a risk-limiting thing because we were late in the cycle.

Reply:
Oh right, I just remembered. We don't allow any intrinsic pregeneration outside CoreLib because the way we do it for CoreLib is that we prevent intrinsic expansion of the `IsSupported` checks and make that check JITted.

This works great for code that properly guards things with `IsSupported`. But code that doesn't guard things with `IsSupported` would get illegal instruction traps instead of `PlatformNotSupportedException` when the CPU doesn't support it. Those are harder to debug so we limit the risk of customers running into that in their code by just making everything jitted.

Reply:
Thanks for the explanation, it makes sense. I tried to summarize it as succinctly as I can, can you please take a look again?